config/langs.toml

```diff
@@ -541,6 +541,23 @@ ForEach ($arg in $args) {
 }
 '''
 
+[Prolog]
+size    = '29.5 MiB'
+version = 'SWI-Prolog 8.4.0'
+website = 'https://www.swi-prolog.org/'
+example = '''
+% Printing
+:- writeln("Hello, world").
+
+% Looping
+:- numlist(0, 9, List),
+   maplist(writeln, List).
+
+% Accessing arguments
+:- current_prolog_flag(argv, [_| Args]),
```

Comment:
Not sure if we should be discarding the first argument here.

Reply:
Same as for some other langs, first argument is always `-`.

Reply:
I now believe that we made a mistake here in the way we set this up. The only reason that it receives "-" as the first argument is that we explicitly provide that as the first argument. In normal `swipl` usage, the filename is not provided as the first argument. Therefore, we should do the same. We made the same mistake for Haskell, but I think it's probably too late to fix the problem for Haskell. I don't think it's too late to fix it for Prolog. Also, we can change the IOS `current_prolog_flag` to simply `prolog_flag`.

Reply:
To be clear, I'm suggesting that we either add an explicit special case for Prolog in `hole/play.go` and remove the `-` argument, or modify the `prolog` script by adding `shift` to consume the parameter. The latter approach is probably better.